File: CS/07_Attachments/GetAllAttachments.cs
﻿using Spire.Pdf;
using Spire.Pdf.Attachments;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace GetAllAttachments
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Create a new PDF document.
            PdfDocument pdf = new PdfDocument();

            //Load the file from disk.
            pdf.LoadFromFile(@"..\..\..\..\..\..\Data\Template_Pdf_2.pdf");

            //Get a collection of attachments on the PDF document.
            PdfAttachmentCollection collection = pdf.Attachments;

            //Save all the attachments to the files.
            for (int i = 0; i < collection.Count; i++) 
            {
                File.WriteAllBytes(collection[i].FileName, collection[i].Data);
            }
        }
    }
}

File: VB/07_Attachments/GetAllAttachments.vb
Imports System.IO
Imports Spire.Pdf
Imports Spire.Pdf.Attachments

Namespace GetAllAttachments
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub
        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Create a new PdfDocument object 
            Dim pdf As New PdfDocument()

            ' Load the PDF document from the specified input file
            pdf.LoadFromFile("..\..\..\..\..\..\Data\Template_Pdf_2.pdf")

            ' Get the collection of attachments from the PDF document
            Dim collection As PdfAttachmentCollection = pdf.Attachments

            ' Iterate through each attachment in the collection
            For i As Integer = 0 To collection.Count - 1
                ' Write the data of the attachment to a file using its original filename
                File.WriteAllBytes(collection(i).FileName, collection(i).Data)
            Next i

            ' Close the PDF document
            pdf.Close()
        End Sub
    End Class
End Namespace
